### ISSUE TITLE
Firestore Rules are beign replaced while sending Push Notifications

### ISSUE THREAD

**Issue by Xefrok** [FlutterFlow/flutterflow-issues#4940]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Hello FlutterFlow Support Team,

I’ve encountered a critical issue with the Push Notifications feature in the FlutterFlow Settings. Each time I send a push notification using this option, it completely overwrites my existing Firestore rules with a different set of rules.

Here are my Firestore rules before clicking on "Send Push Notification":
![image](https://github.com/user-attachments/assets/9e2a8fca-3974-4b8b-9493-778b7e92a518)

And here are my Firestore rules after clicking "Send Push Notification":
![image](https://github.com/user-attachments/assets/b3f17f47-5e71-469c-a671-53ea61ca3620)

This issue caused a significant outage in my production environment for over 6 hours until I identified that the overwritten rules were the root cause. This behavior is not only inconvenient but also poses a severe risk to live applications.

Could you please address this issue urgently? It’s crucial for the Push Notifications feature to work without altering Firestore rules, or at least notify users about any changes it will make.

### Expected Behavior

My rules should not beign changed without my permission or by any other actions rather than the "Deploy Firestore Rules"

### Steps to Reproduce

1.- Create a new blank project
2.- Connect it to firebase
3.- Create 2 or 3 collections and deploy rules
4.- Enable push notifications
5.- Send push notification

### Reproducible from Blank

- [ ] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

Not neccesary

### Visual documentation

![image](https://github.com/user-attachments/assets/f838c8d3-03a0-4dc4-abae-e19daff75972)

### Environment

```markdown
- FlutterFlow version: 5.0.10+
- Platform: Web
- Browser name and version: Chrome
- Operating system and version affected: Windows 11
```

### Additional Information

FYI: I already re-deployed Push Notifications in case was that but in the Cloud Function code it doesn't seems to be the problem i think is triggered from the Send Push Notification button

**Comment by Alezanello** [comment]
Hello,

Thank you for bringing this issue to our attention, and I sincerely apologize for the inconvenience this has caused. I completely understand how impactful this must have been for you.

I’ll escalate this to our engineering team with high priority to ensure it’s addressed as soon as possible.

Thank you for your patience and understanding!

**Comment by aemelyanovff** [comment]
@Xefrok Does your Firebase project have more than one Firestore database in it?

**Comment by Xefrok** [comment]
> @Xefrok Does your Firebase project have more than one Firestore database in it?

Yes, do you think that could generated the issue? but I didn't add the second database to the FlutterFlow project 

![image](https://github.com/user-attachments/assets/934fcb72-8cb2-4b5e-b04e-e15feea1dfa1)

**Comment by aemelyanovff** [comment]
> > @Xefrok Does your Firebase project have more than one Firestore database in it?
> 
> Yes, do you think that could generated the issue? but I didn't add the second database to the FlutterFlow project
> 
> ![image](https://private-user-images.githubusercontent.com/3920698/388634764-934fcb72-8cb2-4b5e-b04e-e15feea1dfa1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyMTM0NDQsIm5iZiI6MTczMjIxMzE0NCwicGF0aCI6Ii8zOTIwNjk4LzM4ODYzNDc2NC05MzRmY2I3Mi04Y2IyLTRiNWUtYjA0ZS1lMTVmZWVhMWRmYTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjFUMTgxOTA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGU3YjY4NGE4Njg3NjczNjYxYzZiMjExNDBmMDZlZGIyMjI0NDJkNTI1NzZlMmEyNWQwZjU3NjBjMWUyZmNhZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.vmjdqw56zsqhJPREG4GpGDoXPHmd5ODvDWQGPWGb6-E)

Thanks for the confirmation.

This is what FlutterFlow does before it sends a push notification:
1. Load existing security rules in your Firebase project.
2. Generate a new rule needed to send the notification.
3. Merge the new rule with existing rules.
4. Deploy rules.

The problem happens on the first step - instead of loading rules for the default database, it loads rules for the non-default database. I believe that this is a bug in Firebase Tools (the utility FlutterFlow uses to interact with user projects). I [filed a bug there](https://github.com/firebase/firebase-tools/issues/7981). In the meantime, we'll try to find some woraround.
